doc/ws.md

```diff
@@ -316,12 +318,23 @@ This class represents a WebSocket. It extends the `EventEmitter`.
     Options given do not have any effect if parsed from the URL given with the
     `address` parameter.
 
+Create a new WebSocket instance.
+
 `perMessageDeflate` default value is `true`. When using an object, parameters
 are the same of the server. The only difference is the direction of requests.
 For example, `serverNoContextTakeover` can be used to ask the server to disable
 context takeover.
 
-Create a new WebSocket instance.
+`finishRequest` is called with arguments
+
+- `request` {http.ClientRequest}
+- `websocket` {WebSocket}
+
+for each HTTP GET request (the initial one and any caused by redirects) when it
+is ready to be sent. If `finishRequest` is set then it has the responsibility to
+call `request.end()` when it is done customizing the request. This is intended
+for niche use-cases that need to defer finishing the request until its socket is
+available, e.g. to add an authentication header involving TLS channel binding.
```

Comment:
Please clarify that customization should be limited to inspecting and adding headers. For example, the user should not add data to the request body or call `request.destroy()`.